### ISSUE TITLE
Regex not matching whitespaces after colon (pretty printed JSON)

### ISSUE THREAD

**Issue by Dorifor** [Ivan-Korolenko/json-with-bigint#10]
Your regex for the JSONParse method does not matches that kind of format: 

```json
[
    {
        "ID": 1035342667379599058,
        "Timestamp": "2022-09-13 22:21:25",
        "Contents": "broken example message",
        "Attachments": ""
    }
]
```

### Proposal

I added `|[^\\]":\s*` to match any number of whitespace after the colon as a possible match inside your regex.

Complete : `/(?<=[^\\]":[\[]?|[^\\]":\[.*[^\.\d*]|[^\\]":\s*)(-?\d{17,}|-?(?:[9](?:[1-9]07199254740991|0[1-9]7199254740991|00[8-9]199254740991|007[2-9]99254740991|007199[3-9]54740991|0071992[6-9]4740991|00719925[5-9]740991|007199254[8-9]40991|0071992547[5-9]0991|00719925474[1-9]991|00719925474099[2-9])))(?=,|\}[^"]?|\][^"])/g`

It works fine with it (with my example value)
### Disclaimer
I'm not really knowledgeable about regex so if you think of a better way I'd love to know.

**Comment by Ivan-Korolenko** [comment]
Hi @Dorifor,

Pretty-printed JSON is not something you usually work with as a value for parsing. It's commonly used only for printing some values, as the name suggests. So I don't see a real-world use case for the support of this feature. 

Furthermore, it's actually pretty hard to implement without breaking the parsing logic of everything else. Solutions I tried out of interest for this topic turned out to be very fragile, especially for complex cases, like string values with long text containing arrays containing big numbers (shouldn't be parsed as BigInt), negative floating point numbers, etc. But even the simple example you provided can be broken very easily:

```
[
    {
        "ID": 1035342667379599058,
        "Timestamp": "2022-09-13 22:21:25",
        "Contents": "broken example message",
        "Attachments": "",
        "BreakingValue": 54354654654654654654656546546546546
    }
]
```

The last value should be parsed as BigInt, but it's not.

You can fix that example by simply adding `[\n]*[\s]*` before the `\}` in the lookahead part of the regular expression. But then what if I do this:

```
{
    "test": [
        {
            "ID": 1035342667379599058,
            "Timestamp": "2022-09-13 22:21:25",
            "Contents": "broken example message",
            "Attachments": "",
            "BreakingValue": 54354654654654654654656546546546546
        },
        1035342667379599058
    ]
}
```

The last value of the array should be parsed as BigInt, but it's not. Etc.

As you can see, this adds a whole layer of complex problems to solve without bringing any real-world usefulness, so unfortunately I can't accept this proposal. Nevertheless, thank you for this proposal. It was interesting to think about this.

Thank you for your interest in the `json-with-bigint` library. Feel free to create a new issue or a PR if you encounter any problems with the library in the future or if you would like to propose some feature or a change.

**Comment by Dorifor** [comment]
I see, there's a lot of edge cases I didn't think of, thanks for the explanation ! 

My use case (Discord messages export) has the json pretty printed by default, so I have to consider it, this repo saved me a lot of headaches !

**Comment by Ivan-Korolenko** [comment]
Sure, no problem. Great to hear that this library helped you!

By the way, I decided to tinker a bit more with pretty printed JSON and eventually I was able to solve this riddle, including all edge cases. 

Now `json-with-bigint` supports pretty printed JSON parsing! **It's implemented in version 2.4.0** and already available on npm.
